Sources/Components/CollapseView/CollapseView.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public protocol CollapseViewDelegate: AnyObject {
+    func didExpand(_ view: CollapseView)
+    func didCollapse(_ view: CollapseView)
+}
+
+public class CollapseView: UIView {
+    // MARK: Public properties
+
+    public weak var delegate: CollapseViewDelegate?
+
+    // MARK: Private properties
+
+    private enum State {
+        case expanded
+        case collapsed
+    }
+
+    private lazy var hairlineView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .bgPrimary
+        view.dropShadow(color: .black, opacity: 0.6, offset: CGSize(width: 0, height: 0.9), radius: 1.2)
+        return view
+    }()
+
+    private lazy var selectorTitleView: SelectorTitleView = {
+        let view = SelectorTitleView(withAutoLayout: true)
+        view.updateButtonColor(.textSecondary, buttonDisabledColor: .textSecondary)
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var contentView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .bgPrimary
+        return view
+    }()
+
+    private var state: State
+
+    private let defaultHeight: CGFloat = 32.0
+    private let defaultContentViewHeight: CGFloat = 32.0
+
+    private var heightAnchorConstraint: NSLayoutConstraint?
+
+    // MARK: Parameters
+
+    private var collapsedTitle: String
+    private var expandedTitle: String
+    private var injectedView: UIView?
+    private var injectedViewHeight: CGFloat = 0.0
+
+    // MARK: Initalizer
+
+    public init(collapsedTitle: String, expandedTitle: String, presentViewInExpandedState: UIView?, heightOfView: CGFloat, withAutoLayout: Bool = false) {
```

Comment:
Don't you think the `presentViewInExpandedState` parameter sounds like a `Bool`? Maybe: `viewToPresentInExpandedState` or similar?